vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -74,37 +47,76 @@ ChakraRuntime::ChakraRuntime(ChakraRuntimeArgs &&args) noexcept : m_args{std::mo
         JsRuntimeAttributeDisableExecutablePageAllocation);
   }
 
-  VerifyChakraErrorElseThrow(JsCreateRuntime(runtimeAttributes, nullptr, &m_runtime));
+  ChakraVerifyJsErrorElseThrow(JsCreateRuntime(runtimeAttributes, nullptr, &m_runtime));
 
   setupMemoryTracker();
 
-  JsContextRef context = JS_INVALID_REFERENCE;
-  VerifyChakraErrorElseThrow(JsCreateContext(m_runtime, &context));
-  m_context.Initialize(context);
+  m_context = JsRefHolder{CreateContext(m_runtime)};
 
   // Note :: We currently assume that the runtime will be created and
   // exclusively used in a single thread.
-  VerifyChakraErrorElseThrow(JsSetCurrentContext(m_context));
+  // Preserve the current context if it is already associated with the thread.
+  m_prevContext = JsRefHolder{GetCurrentContext()};
+  SetCurrentContext(m_context);
 
   startDebuggingIfNeeded();
 
   setupNativePromiseContinuation();
 
   std::call_once(s_runtimeVersionInitFlag, initRuntimeVersion);
 
-  static facebook::jsi::StringBuffer bootstrapBundleSourceBuffer{g_bootstrapBundleSource};
-  evaluateJavaScriptSimple(bootstrapBundleSourceBuffer, "ChakraRuntime_bootstrap.bundle");
+  m_propertyId.Object = JsRefHolder{GetPropertyIdFromName(L"Object")};
```

Comment:
>Object [](start = 60, length = 6)

consider giving names to these property names esp if you're using them in another place to prevent typos etc

Reply:
This is exactly the goal of the m_propertyId here: we assign a field name for each string for efficiency.

---
In reply to: [470465553](https://github.com/microsoft/react-native-windows/pull/5710#discussion_r470465553) [](ancestors = 470465553)